### PR TITLE
Destroy Env script enhancements

### PR DESCRIPTION
**Pull request by oren-zohar**
### Summary of your changes
Small `Destroy Env` script changes:
- Linting
- Ensure it exits with an error if one of the Envs failed to delete

### CODE REVIEW

File: deploy/test-environments/delete_env.sh [hunk 1]
```diff
@@ -24,14 +24,14 @@ function delete_environment() {
     tfstate="./$ENV-terraform.tfstate"
 
     # Copy state file
-    if aws s3 cp $BUCKET/"$ENV"/terraform.tfstate "$tfstate"; then
+    if aws s3 cp "$BUCKET"/"$ENV"/terraform.tfstate "$tfstate"; then
```
Comment (orestisfl) [inline deploy/test-environments/delete_env.sh:27]:
```suggestion
    if aws s3 cp "$BUCKET/$ENV/terraform.tfstate" "$tfstate"; then
```

Slightly more readable IMO

File: deploy/test-environments/delete_env.sh [hunk 2]
```diff
@@ -24,14 +24,14 @@ function delete_environment() {
...
         echo "Downloaded Terraform state file from S3."
 
         # Check if the resource aws_auth exists in the local state file and remove it
         terraform state rm -state "$tfstate" $(terraform state list -state "$tfstate" | grep "kubernetes_config_map_v1_data.aws_auth") || true
         # Destroy environment and remove environment data from S3
         if terraform destroy -var="region=$AWS_REGION" -state "$tfstate" --auto-approve &&
-            aws s3 rm $BUCKET/"$ENV" --recursive; then
+            aws s3 rm "$BUCKET"/"$ENV" --recursive; then
```
Comment (orestisfl) [inline deploy/test-environments/delete_env.sh:34]:
```suggestion
            aws s3 rm "$BUCKET/$ENV" --recursive; then
```

**Review by orestisfl — COMMENTED** [review]
While you are at it, can you also add the script in the pre-commit hook for shellcheck?

**Review by gurevichdmitry — APPROVED** [review]
@oren-zohar, LGTM

**Review by orestisfl — APPROVED** [review]

**Comment by oren-zohar** [comment]
I'll run it on the branch before merging, please don't merge

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -77,7 +77,7 @@ repos:
       - id: shfmt
         args: ["-i", "4"]
       - id: shellcheck
-        files: (deploy/azure/.*sh|scripts/publish_cft.sh|.ci/scripts/.*sh)
+        files: (deploy/azure/.*sh|scripts/publish_cft.sh|.ci/scripts/.*sh|deploy/test-environments/.*sh)
```
Comment (orestisfl) [inline .pre-commit-config.yaml:80]:
```suggestion
        files: (deploy/.*sh|scripts/publish_cft.sh|.ci/scripts/.*sh)
        exclude: deploy/deployment-manager/.*sh
```

**Pull request merged**